camel/synthetic_datagen/method_factory.py

```diff
@@ -0,0 +1,64 @@
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+from enum import Enum
+from typing import Any, Type, Union
+
+from .base_generator import BaseDataGenerator
+from .self_instruct.self_instruct_pipeline import (
+    SelfInstructPipeline,
+)
+
+
+class SyntheticDataGeneratorMethodType(Enum):
+    SELFINSTRUCT = "selfinstruct"
+
+    @property
+    def is_self_instruct(self) -> bool:
+        return self is SyntheticDataGeneratorMethodType.SELFINSTRUCT
```

Comment:
why we need to check this? I think we can do the check by using .value

Reply:
@andrei3131